packages/features/form-builder/FormBuilder.tsx

```diff
@@ -0,0 +1,737 @@
+import { useAutoAnimate } from "@formkit/auto-animate/react";
+import { ErrorMessage } from "@hookform/error-message";
+import { useState } from "react";
+import { Controller, useFieldArray, useForm, useFormContext } from "react-hook-form";
+import { z } from "zod";
+
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import slugify from "@calcom/lib/slugify";
+import {
+  Label,
+  Badge,
+  Button,
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogHeader,
+  DialogFooter,
+  Form,
+  BooleanToggleGroupField,
+  SelectField,
+  InputField,
+  Input,
+  showToast,
+} from "@calcom/ui";
+import { Switch } from "@calcom/ui";
+import { FiArrowDown, FiArrowUp, FiX, FiPlus, FiTrash2, FiInfo } from "@calcom/ui/components/icon";
+
+import { Components } from "./Components";
+import { fieldsSchema } from "./FormBuilderFieldsSchema";
+
+type RhfForm = {
+  fields: z.infer<typeof fieldsSchema>;
+};
+
+type RhfFormFields = RhfForm["fields"];
+type RhfFormField = RhfFormFields[number];
+
+/**
+ * It works with a react-hook-form only.
+ * `formProp` specifies the name of the property in the react-hook-form that has the fields. This is where fields would be updated.
+ */
+export const FormBuilder = function FormBuilder({
+  title,
+  description,
+  addFieldLabel,
+  formProp,
+}: {
+  formProp: string;
+  title: string;
+  description: string;
+  addFieldLabel: string;
+}) {
+  const FieldTypes: {
+    value: RhfForm["fields"][number]["type"];
+    label: string;
+    needsOptions?: boolean;
+    systemOnly?: boolean;
+    isTextType?: boolean;
+  }[] = [
+    {
+      label: "Name",
+      value: "name",
+      isTextType: true,
+    },
+    {
+      label: "Email",
+      value: "email",
+      isTextType: true,
+    },
+    {
+      label: "Phone",
+      value: "phone",
+      isTextType: true,
+    },
+    {
+      label: "Short Text",
+      value: "text",
+      isTextType: true,
+    },
+    {
+      label: "Number",
+      value: "number",
+      isTextType: true,
+    },
+    {
+      label: "Long Text",
+      value: "textarea",
+      isTextType: true,
+    },
+    {
+      label: "Select",
+      value: "select",
+      needsOptions: true,
+      isTextType: true,
+    },
+    {
+      label: "MultiSelect",
+      value: "multiselect",
+      needsOptions: true,
+      isTextType: false,
+    },
+    {
+      label: "Multiple Emails",
+      value: "multiemail",
+      isTextType: true,
+    },
+    {
+      label: "Radio Input",
+      value: "radioInput",
+      isTextType: false,
+      systemOnly: true,
+    },
+    {
+      label: "Checkbox Group",
+      value: "checkbox",
+      needsOptions: true,
+      isTextType: false,
+    },
+    {
+      label: "Radio Group",
+      value: "radio",
+      needsOptions: true,
+      isTextType: false,
+    },
+    {
+      label: "Checkbox",
+      value: "boolean",
+      isTextType: false,
+    },
+  ];
```

Comment:
I am not sure if we can control the order of these types reliably now(across browsers).

Reply:
Okay, I just read that the order is reliable since many years now. So, if property names are strings, they would be iterated in the order they are created. So this should be fine now.